File: Python3/0594_longest_harmonious_subsequence.py
# https://leetcode-cn.com/problems/longest-harmonious-subsequence/
class Solution:
    def findLHS(self, nums):
        """
        Inputs:
            nums: list[int]
        Outputs:
            res: int
        """

        count = {}
        for i in nums:
            count[i] = count.get(i,0) + 1

        res = 0
        for i in count:
            tmp = count[i]
            res = max(res, tmp + count.get(i+1,-tmp))
        return res

File: Cpp/0594_longest_harmonious_subsequence.cpp
// https://leetcode.cn/problems/longest-harmonious-subsequence/

class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> cnt;
        for (int num: nums) {
            cnt[num] += 1;
        }

        int res = 0;
        for (auto it = cnt.begin(); it != cnt.end(); ++it) {
            int num = it->first, c = it->second;
            if ((cnt.find(num+1) != cnt.end()) && (cnt[num+1] + c > res)) {
                res = cnt[num+1] + c;
            }
        }
        return res;
    }
};
